project/apps/web/app/questions/components/CreateModal.tsx

```diff
@@ -0,0 +1,201 @@
+                  </FormControl>
+                  <FormMessage />
+                </FormItem>
+              )}
+            />
+
+            {/* Description Field */}
+            <FormField
+              control={form.control}
+              name="q_desc"
+              render={({ field }) => (
+                <FormItem>
+                  <FormLabel className="text-black">Description</FormLabel>
+                  <FormControl>
+                    <Textarea
+                      className="resize-y max-h-64"
+                      placeholder="Enter description"
+                      spellCheck={false}
+                      {...field}
+                    />
+                  </FormControl>
+                  <FormMessage />
+                </FormItem>
+              )}
+            />
+
+            {/* Complexity Dropdown */}
+            <FormField
+              control={form.control}
+              name="q_complexity"
+              render={({ field }) => (
+                <FormItem>
+                  <FormLabel className="text-black">Complexity</FormLabel>
+                  <FormControl>
+                    <Select value={field.value} onValueChange={field.onChange}>
+                      <SelectTrigger>
+                        <SelectValue placeholder="Select complexity" />
+                      </SelectTrigger>
+                      <SelectContent>
+                        {Object.values(COMPLEXITY).map((level) => (
+                          <SelectItem key={level} value={level}>
+                            {level}
+                          </SelectItem>
+                        ))}
+                      </SelectContent>
+                    </Select>
+                  </FormControl>
+                  <FormMessage />
+                </FormItem>
+              )}
+            />
+
+            {/* Categories Multi-Select */}
+            <FormField
+              control={form.control}
+              name="q_category"
+              render={() => (
+                <FormItem>
+                  <FormLabel className="text-black">Categories</FormLabel>
+                  <FormControl>
+                    <div className="flex flex-wrap gap-2">
+                      {categories.map((cat) => (
+                        <Badge
+                          key={cat}
+                          onClick={() => toggleCategory(cat)}
+                          variant={
+                            selectedCategories.includes(cat)
+                              ? "default"
+                              : "secondary"
+                          }
+                          className="cursor-pointer"
+                        >
+                          {cat}
+                        </Badge>
+                      ))}
+                    </div>
+                  </FormControl>
+                  <FormMessage />
+                </FormItem>
+              )}
+            />
```

Comment:
would suggest sticking to use the form's values instead of a separate `useState` variable, so that the form has full control.
```suggestion
<FormField
              control={form.control}
              name="q_category"
              render={({ field }) => (
                <FormItem>
                  <FormLabel className="text-black">Categories</FormLabel>
                  <FormControl>
                    <div className="flex flex-wrap gap-2">
                      {categories.map((cat) => (
                        <Badge
                          key={cat}
                          onClick={() =>
                            field.value?.includes(cat)
                              ? field.onChange(
                                  field.value?.filter((value) => value !== cat),
                                )
                              : field.onChange([...field.value, cat])
                          }
                          variant={
                            field.value?.includes(cat) ? "default" : "secondary"
                          }
                          className="cursor-pointer"
                        >
                          {cat}
                        </Badge>
                      ))}
                    </div>
                  </FormControl>
                  <FormMessage />
                </FormItem>
              )}
            />
```